ecl/hql/hqlgram.y

```diff
@@ -3079,6 +3079,79 @@ hintExpr
                         }
     ;
 
+orderAttribute
+    : UNORDERED
+                        {
+                            $$.setExpr(createExprAttribute(orderedAtom, createConstant(false)), $1);
+                        }
+    | ORDERED
+                        {
+                            $$.setExpr(createExprAttribute(orderedAtom), $1);
+                        }
+    | ORDERED '(' expression ')'
+                        {
+                            parser->normalizeExpression($3, type_boolean, true);
+                            $$.setExpr(createExprAttribute(orderedAtom, $3.getExpr()), $1);
+                        }
+    | PARALLEL
+                        {
+                            $$.setExpr(createExprAttribute(parallelAtom), $1);
+                        }
+    | PARALLEL '(' expression ')'
+                        {
+                            parser->normalizeExpression($3, type_int, true);
+                            $$.setExpr(createExprAttribute(parallelAtom, $3.getExpr()), $1);
+                        }
+    | STABLE
+                        {
+                            $$.setExpr(createExprAttribute(stableAtom));
+                            $$.setPosition($1);
+                        }
+    | STABLE '(' expression ')'
+                        {
+                            parser->normalizeExpression($3, type_string, false);
+                            $$.setExpr(createExprAttribute(stableAtom, $3.getExpr()));
+                            $$.setPosition($1);
+                        }
+    | UNSTABLE
+                        {
+                            $$.setExpr(createExprAttribute(unstableAtom));
+                            $$.setPosition($1);
+                        }
+    | UNSTABLE '(' expression ')'
+                        {
+                            parser->normalizeExpression($3, type_string, false);
```

Comment:
It's a string for historical reasons, because we support putting the name of the sort algorithm in there, I think. We need to keep it that way for backward compatibility.